### PR TITLE
Fix #579 : Use spring boot dependencies bom instead of spring boot pa…

### LINKED ISSUE #579: Using Spring Boot without the Parent POM
Sugestion: Change spring-boot-starter-parent to dependency management,

Allowing greater decoupling between the software and the framework

When migrating to other framework like quarkus it would be simpler

https://docs.spring.io/spring-boot/docs/current/maven-plugin/reference/htmlsingle/#using.import

### PR DESCRIPTION
**Pull request by beatfreaker**
related to #579

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
@beatfreaker : thanks for your contribution. Can you rebase against main branch plz? Just did a quick fix, related to liquibase

File: src/main/java/tech/jhipster/lite/generator/server/springboot/core/domain/SpringBootDomainService.java [hunk 1]
```diff
@@ -99,6 +101,21 @@ public void addSpringBootMavenPlugin(Project project) {
             .groupId(SPRINGBOOT_PACKAGE)
             .artifactId("spring-boot-maven-plugin")
             .version("\\${spring-boot.version}")
+            .additionalElements(
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/core/domain/SpringBootDomainService.java:104]: Unless I'm wrong, this part should not change. You need to add another part for plugin management

You can have a look at:
- https://github.com/jhipster/jhipster-sample-app/blob/main/pom.xml#L354-L357
- https://github.com/jhipster/jhipster-sample-app/blob/main/pom.xml#L764-L784

Comment (beatfreaker) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/core/domain/SpringBootDomainService.java:104]: Got it, and adding that logic within `addSpringBootMavenPlugin()` is good, or should we create new method specifically for adding plugin into `<pluginManagement>` ?

Comment (beatfreaker) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/core/domain/SpringBootDomainService.java:105]: Didn't add below comment, is it fine ? or we should add it ?

```xml
<!--
    Enable the line below to have remote debugging of your application on port 5005
    <jvmArguments>-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005</jvmArguments>
-->
```

Comment (pascalgrimaud) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/core/domain/SpringBootDomainService.java:105]: No need I think. Never used it myself :-)

**Comment by pascalgrimaud** [comment]
I'll review and test it tomorrow :-)

File: src/main/java/tech/jhipster/lite/generator/server/springboot/core/domain/SpringBootDomainService.java [hunk 2]
```diff
@@ -99,16 +102,36 @@ public void addSpringBootMavenPlugin(Project project) {
...
+              """
+                    <executions>
+                      <execution>
+                          <goals>
+                              <goal>repackage</goal>
+                          </goals>
+                      </execution>
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/core/domain/SpringBootDomainService.java:112]:
```suggestion
                      <execution>
                        <goals>
                          <goal>repackage</goal>
                        </goals>
                      </execution>
```

File: src/main/java/tech/jhipster/lite/generator/server/springboot/core/domain/SpringBootDomainService.java [hunk 3]
```diff
@@ -43,20 +44,22 @@ public void init(Project project) {
...
         version -> {
           project.addConfig("springBootVersion", version);
 
-          Parent parent = Parent
+          Dependency springBootDependenciesPom = Dependency
             .builder()
             .groupId(SPRINGBOOT_PACKAGE)
-            .artifactId("spring-boot-starter-parent")
+            .artifactId("spring-boot-dependencies")
             .version((String) project.getConfig("springBootVersion").orElse(version))
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/core/domain/SpringBootDomainService.java:57]:
```suggestion
            .version("\\${spring-boot.version}")
```

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Comment by pascalgrimaud** [comment]
@beatfreaker : I think the test needs to be updated too ;)

**Comment by beatfreaker** [comment]
> @beatfreaker : I think the test needs to be updated too ;)

sorry forgot to test it, as tried to provide quick update using `github.dev web editor` my bad.

**Review by pascalgrimaud — APPROVED** [review]

**Comment by pascalgrimaud** [comment]
Thanks and congrats for your 1st contribution @beatfreaker !

**Comment by beatfreaker** [comment]
Bounty claimed at https://opencollective.com/generator-jhipster/expenses/62764

**Comment by pascalgrimaud** [comment]
@beatfreaker : approved

**Pull request merged**